src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Parser.cs

```diff
@@ -551,23 +553,28 @@ private bool TryGetTypeSpec(ITypeSymbol type, DiagnosticDescriptor descriptor, o
                 return true;
             }
 
-            private ArraySpec? CreateArraySpec(IArrayTypeSymbol arrayType, Location? location)
+            private EnumerableSpec? CreateArraySpec(IArrayTypeSymbol arrayType, Location? location)
             {
                 if (!TryGetTypeSpec(arrayType.ElementType, Helpers.ElementTypeNotSupported, out TypeSpec elementSpec))
                 {
                     return null;
                 }
 
                 // We want a BindCore method for List<TElement> as a temp holder for the array values.
-                EnumerableSpec? listSpec = ConstructAndCacheGenericTypeForBindCore(_typeSymbols.List, arrayType.ElementType) as EnumerableSpec;
+                EnumerableSpec? listSpec = ConstructGenericTypeSpec(_typeSymbols.List_Unconstructed, arrayType.ElementType) as EnumerableSpec;
                 // We know the element type is supported.
                 Debug.Assert(listSpec != null);
+                if (listSpec is not null)
+                {
+                    AddToRootConfigTypeCache(MethodSpecifier.BindCore, listSpec);
+                }
 
-                return new ArraySpec(arrayType)
+                return new EnumerableSpec(arrayType)
                 {
                     Location = location,
```

Comment:
If you're planning on making those types cache-friendly, you should consider trimming the `Location` instances you're passing on since they're capturing a reference to the `Compilation` object.

See https://github.com/dotnet/runtime/pull/83868 for more details.

Reply:
Will address when fixing https://github.com/dotnet/runtime/issues/83534.